### PR TITLE
Implement shareReplay, switch form to async pipe

### PR DESCRIPTION
**Pull request by emcapone**
I've set up shareReplay so that Pets is cached rather than making a get request on every page load. This is only refreshed on actions that would update the data. It is called in the subscription and not the service. I also changed the form to accept a Pet so that the pet form could use the async pipe.
I'm putting in a PR now without many changes because I'm going to switch to working with some more Material and CSS for a bit. Please do let me know if I am not using best practices or if you'd suggest any other changes. Thanks!

### CODE REVIEW

File: src/app/pet.service.ts [hunk 1]
```diff
@@ -17,12 +17,20 @@ export class PetService {
...
 
-  pets$ = this.http.get<Pet[]>(this.petsUrl)
+  private _petsData$ = new BehaviorSubject<void>(undefined);
+  apiRequest$ = this.http.get<Pet[]>(this.petsUrl)
     .pipe(
-      tap(_ => console.log('fetched pets')),
-      catchError(this.handleError<Pet[]>('fetch pets', []))
+      tap(pet => {
+        console.log('fetch pets');
+        return pet;
```
Comment (swhook52) [inline src/app/pet.service.ts]: a `tap` is a side effect. Something that happens passively. You do not need to return anything from a tap.

File: src/app/edit-pet/edit-pet.component.html [hunk 2]
```diff
@@ -1,10 +1,9 @@
 <div>
   <h1>Edit Pet</h1>
   <hr>
-  <div class="spinner" *ngIf="!this.loaded">
-    <mat-spinner></mat-spinner>
+  <div *ngIf="this.pet$">
```
Comment (swhook52) [inline src/app/edit-pet/edit-pet.component.html]: The observable will always be truthy

**Review by swhook52 — CHANGES_REQUESTED** [review]
Just a couple of small changes. I'm happy to see you use the BehaviorSubject pattern to control your own observables.

File: src/app/pet.service.ts (hunk 1 above)
Comment (emcapone) [reply src/app/pet.service.ts]: I removed the return and modified the console message.

File: src/app/edit-pet/edit-pet.component.html (hunk 2 above)
Comment (emcapone) [reply src/app/edit-pet/edit-pet.component.html]: I just removed the div entirely since it isn't doing anything.

**Pull request merged**